### PR TITLE
add support for 'Member of List' & 'Remove Leads from List'

### PR DESCRIPTION
**Pull request by cgrayson**
As well as a convenience method to find a single list by ID.

### CODE REVIEW

File: lib/api/list.js [hunk 1]
```diff
@@ -63,6 +75,17 @@ List.prototype = {
       _method: 'GET'
     });
     return this._connection.post(path, {data: options});
+  },
+
+  isMember: function(listId, leadIds) {
+    var path = util.createPath('lists', listId, 'leads', 'ismember.json');
+    leadIds = [].concat(leadIds); // ensure it's an array
+    return this._connection.get(path + "?id=" + leadIds.join('&id='));
```
Comment (peaches) [inline lib/api/list.js]: The `get` call accepts a second parameter for the query parameters to make it more consistent with the rest of the code.

```js
this._connection.get(path, {id: leadIds.join('&id=')})
```

File: lib/api/list.js [hunk 2]
```diff
@@ -43,6 +43,18 @@ List.prototype = {
...
+  removeLeadsFromList: function(listId, leadIds) {
+    if (!_.isArray(leadIds) && !_.isEmpty(leadIds)) {
+      var msg = 'input must be an array of lead ids';
+      log.error(msg);
+      return Promise.reject(msg);
+    }
+
+    var path = util.createPath('lists', listId, 'leads.json');
+    leadIds = [].concat(leadIds); // ensure it's an array
+    return this._connection.del(path + "?id=" + leadIds.join('&id='), {headers: {"Content-Type": "application/json"}});
```
Comment (peaches) [inline lib/api/list.js]: Same with the other comment, can you see if it is possible to combine the query parameters with the headers into a single object?

```js
this._connection.del(path, {id: leadIds.join('&id='), headers: ...});
```

In addition, I've been using single quotes `'` in this code unless I'm quoting something that contains a single quote.

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "node-marketo",
-  "version": "0.2.4",
+  "version": "0.3.1",
```
Comment (peaches) [inline package.json]: maybe 0.3.0?

Comment (peaches) [inline package.json]: I can do this on my side when I create a tag for it as well.

File: lib/api/list.js [hunk 4]
```diff
@@ -43,6 +43,18 @@ List.prototype = {
...
 
+  removeLeadsFromList: function(listId, leadIds) {
+    if (!_.isArray(leadIds) && !_.isEmpty(leadIds)) {
+      var msg = 'input must be an array of lead ids';
+      log.error(msg);
+      return Promise.reject(msg);
+    }
+
+    var path = util.createPath('lists', listId, 'leads.json');
+    leadIds = [].concat(leadIds); // ensure it's an array
```
Comment (peaches) [inline lib/api/list.js]: The conditional check above `!_.isArray(leadIds)` will render this call useless.  I like the fact that the API accepts an array, so I think removing this conversion is more ideal and still safe.

File: lib/api/list.js [hunk 5]
```diff
@@ -63,6 +75,17 @@ List.prototype = {
       _method: 'GET'
     });
     return this._connection.post(path, {data: options});
+  },
+
+  isMember: function(listId, leadIds) {
+    var path = util.createPath('lists', listId, 'leads', 'ismember.json');
```
Comment (peaches) [inline lib/api/list.js]: Should we add a check for `leadIds` here as well?

**Comment by peaches** [comment]
This is great! Thanks for taking the time to understand the code. Let me know if something I'm saying doesn't make sense.

Also, thanks for the tests, it's not the most straightforward way of generating data.

**Comment by cgrayson** [comment]
Hi, thanks for the review before. I'm back from vacation (and from digging out after being gone a week :smile: ); here's an update with your feedback. 

Looks like the restler API calls are a little different than you suggested, but those calls look cleaner now. Also reverted the version (I'd bumped that prematurely for use in the module we're building on top of it). 

Let me know what you think. Planning to add support for "Opportunities", "Opportunity Roles", and "Custom Objects, next, if you don't mind.

**Comment by cgrayson** [comment]
Apologies… hold on these latest changes, they're not working.

**Comment by cgrayson** [comment]
Ok, that's better. :-/ Plus, I remembered to fix the tests this time.

**Comment by peaches** [comment]
Thanks for following up, this is great!

**Comment by peaches** [comment]
As far as implementing those other functionalities, that would be great. I don't have the bandwidth to do additional features at the moment but I am happy to accept PRs. Thanks for all the work.

**Comment by peaches** [comment]
I've just tagged this as 0.3.0

**Comment by cgrayson** [comment]
Great, thank you! And I should have been more clear: I intend to add the other features. :-)

**Pull request merged**